be/src/runtime/task_group/task_group.h

```diff
@@ -60,23 +66,40 @@ using TGEntityPtr = TaskGroupEntity*;
 
 class TaskGroup {
 public:
-    TaskGroup(uint64_t id, std::string name, uint64_t cpu_share);
+    TaskGroup(uint64_t id, std::string name, uint64_t cpu_share, int64_t version);
 
     TaskGroupEntity* task_entity() { return &_task_entity; }
 
-    uint64_t share() const { return _share; }
+    uint64_t cpu_share() const {
+        std::shared_lock<std::shared_mutex> rl {mutex};
+        return _cpu_share;
```

Comment:
How about using std::atomic for _cpu_share?
method cpu_share() will be called frequently.
Maybe just keep the eventually consistency?

Reply:
has been replaced with atomic